src/utils/wallet.js

```diff
@@ -314,19 +351,35 @@ class Wallet {
         this.accounts[accountId] = true
     }
 
-    async addAccessKey(accountId, contractId, publicKey) {
-        try {
-            return await this.getAccount(accountId).addKey(
-                publicKey,
-                contractId,
-                '', // methodName
-                ACCESS_KEY_FUNDING_AMOUNT
-            )
-        } catch (e) {
-            if (e.type === 'AddKeyAlreadyExists') {
-                return true;
+
+    /********************************
+    recovering a second account attempts to call this method with the currently logged in account and not the tempKeyStore 
+    ********************************/
+    async addAccessKey(accountId, contractId, publicKey, fullAccess = false) {
+        const { account, has2fa } = await this.getAccountAndState()
+        if (!contractId) {
```

Comment:
Why not let caller set `contractId` as needed? Seems to be less error prone potentially.

Also better not to mix usage of `accountId` as parameter and current `accountId` in wallet.  E.g. `this.getAccountAndState()` relies on current account ID, but following does use passed `accountId` from parameter:

```
return await this.getAccount(accountId).addKey(
                         publicKey,
                         contractId,
                         '', // methodName
                         ACCESS_KEY_FUNDING_AMOUNT
                     )
```

Reply:
cleaned up and disambiguated